File: solutions/3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        int ans = 0; 
        unordered_map<char, int> freq; 
        for (int i = 0, ii = 0; i < s.size(); ++i) {
            ++freq[s[i]]; 
            while (freq[s[i]] == 3) --freq[s[ii++]]; 
            ans = max(ans, i-ii+1); 
        }
        return ans; 
    }
};

File: solutions/3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.cs
public class Solution {
    public int MaximumLengthSubstring(string s) {
        int n = s.Length;
        int res = 0;
        for (int left = 0; left < n; left++) {
            int[] count = new int[26];
            for (int right = left; right < n; right++) {
                int ch = s[right] - 'a';
                count[ch]++;
                if (count[ch] > 2) {
                    break;
                }
                res = Math.Max(res, right - left + 1);
            }
        }
        return res;
    }
}